### ISSUE TITLE
diomedia.com : Invalid XML error

### ISSUE THREAD

**Issue by kaithaba** [lovasoa/dezoomify#39]
http://www.diomedia.com/imageDetails.do?imageId=14321102  Is the page I'm trying to rip the full-size image from. I fed that URL into Dezoomify, but now I'm getting this error: Error: Invalid XML: http://www.diomedia.com/,window.XMLHttpRequest&&XMLHttpRequest.prototype&&XMLHttpRequest.prototype.addEventListener&&window.XML (http://ophir.alwaysdata.net/dezoomify/zoommanager.js:107)

I'm not that code-savvy, so apologies if I'm just dumb; I tried looking into the source code but couldn't find the right .xml file, I guess.

**Comment by lovasoa** [comment]
Looks like they don't include the url to the meta-information file directly in their page. But you can still get the URL, [following the instructions from the FAQ](https://github.com/lovasoa/dezoomify/wiki/Dezoomify-FAQ#the-image-uses-an-image-viewer-supported-by-dezoomify-but-dezoomify-still-doesnt-work-what-to-do-).

In your case, the URL is : http://www.diomedia.com/imgServer/dioSrv.fcgi?deepzoom=6b9bb3a990767040fe670c53d3e51d86.jp2.dzi

I could hardcode support for this site, but I think it's not very interesting, because they anyway add watermarks to their tiles...

**Comment by lovasoa** [comment]
I've just found a bug in their image preview script, that allows you to download the whole image in full resolution without watermark:

 * Find the id of your image. It's in the URL. In your example the url is `http://www.diomedia.com/imageDetails.do?imageId=14321102` so the id is **`14321102`**.
 * Download the image at `http://www.diomedia.com/imagePreview/01AP3VJ9?ds=-1&imageId=$id`, where you replace `$id` with the id you found at step 1. For your example, it gives `http://www.diomedia.com/imagePreview/01AP3VJ9?ds=-1&imageId=14321102`.
  * The `ds=-1` part of the URL triggers a bug in the server. It fails to calculate the size of the image (that is supposed to be just a preview), and returns a really huge image.
   * You can then resize the resulting image, to give it it's original size. You can find the original size in the `.dzi` file I mentioned in my previous comment. In your example, it's Width=5266 and Height=3438.

**Comment by kaithaba** [comment]
I did read the FAQ (always do!), but I still couldn't find the relevant files, because it wasn't where I was told to look, I don't think; or then I was looking in the wrong place. For someone who doesn't usually look deep into the code side of things, it was pretty complicated. But what I lack in coding knowledge, I make up for in Photoshop skillz (enough to know how to dewatermark those kinds of things), so I'm still interested in the images, tagged or untagged!:) But it's fantastic that you managed to find that bug, wow. Thanks a million! 

I know this is closed, but I have one further question, and it's just the thing about finding the .dzi file. Can you give me any hints of alternative ways of finding it on that particular site, since I couldn't find it according to the instructions on the FAQ? (I admit I might have just been too scatterbrained, but it *was* tricky to find). I even went through AdBlock's breakdown of the site's elements (I often use that to scavenge sites for images they're trying to block you from downloading). Just in case they find out about that bug and fix it and it won't work for finding the full-res unwatermarked images anymore. Thanks massively for the tips you've already given me--now I'm off to scavenge as many pics as I can before they fix it!

**Comment by lovasoa** [comment]
I updated the FAQ with more detailed instructions, and screenshots: https://github.com/lovasoa/dezoomify/wiki/Dezoomify-FAQ#the-image-uses-an-image-viewer-supported-by-dezoomify-but-dezoomify-still-doesnt-work-what-to-do-

I hope you find it useful. Tell me if there are still ambiguities.

**Comment by kaithaba** [comment]
Oh yes, the XHR and bringing the zoomable image up things weren't there before; much easier now. Thanks!  Although that particular site's now just giving me 504 timeouts... might be the images just being so damn huge. Was there some proxy thing around this? Although this is going beyond the scope of Dezoomify and I'll understand if you don't have time to go around looking. Thanks massively for your help.

**Comment by lovasoa** [comment]
Yes, the bug generates very large images, that must use up their server's RAM and CPU. I would recommend that you wait a little between file downloads, download just what you need, and if possible do your downloads only at times when the server is not busy. No proxy would help with that.

**Comment by kaithaba** [comment]
Yeah, I figured, but it's odd that I've only ever managed to download that first image so far, and another from the same film, and then everything after that has timed out. I talked about scavenging as many pics as I could, but it was never more than a dozen or so old movie pics I had in mind, so it's crazy that it'd go boom after just two (I tried with an anonymous browser as well, so it can't have been them blocking my IP addy yet or anything). But ah well, thanks again for your time and I'll see how it goes--I bet they'll find out soon enough, but hope I can nick a few more golden oldies before they catch up.
